project/target/Cerca-1/script.js

```diff
@@ -0,0 +1,149 @@
+    });
+}
+
+// Creates a favorites bar element
+function createFavoritesElement(favorite) {
+  placeID = favorite;
+  const favElement = document.createElement('a');
+  favElement.innerText = favorite;
+  favElement.addEventListener("click", openModal(), false);
+  return favElement;
+}
+
+// Modal opens displaying two options: go to restaurant page, or get shareable link
+function openModal () {
+    document.getElementById("favoritesModal").style.display = "block";
+}
+
+// When the user clicks on the modal's X button, close it
+function closeModal () {
+    document.getElementById("favoritesModal").style.display = "none";
+}
+
+// When the user clicks anywhere outside of the modal, close it
+window.onclick = function(event) {
+  var modal = document.getElementById("favoritesModal");
+  if (event.target == modal) {
+    modal.style.display = "none";
+  }
+}
+
+// Go to restaurant page
+function visitPage() {
+    window.location.replace(placeID); 
+}
+
+// Constructs a shortened URL that leads to the chosen restaurant
+function shareableURL() {
+    //const currentURL = window.location.href;
+    const shareableURL = window.location+"?placeID=<placeID>";
+    alert("Shareable link: " + shareableURL);
+}
+
+// Called once user has selected their preferred filters and pressed the Search button
+// Stores filter choices
+function filterChoices() {
+    const radius = document.getElementById("mySelectRadius").selectedIndex;
+    const type = document.getElementById("mySelectType").selectedIndex + 4;
+    const price = document.getElementById("mySelectPrice").selectedIndex + 9;
+    console.log(document.getElementsByTagName("option")[radius].value);
+    console.log(document.getElementsByTagName("option")[type].value);
+    console.log(document.getElementsByTagName("option")[price].value);
+}
+
+// Only display the top 5 most popular restaurants
+function displayPopular() {
+    fetch('/popular').then(response => response.json()).then((popular) => {
+        const popularList = document.getElementById('popular-list');
+        
+        // Get list of restaurants with the 5 highest popularity scores
+        var popScores = [];
+        for (const score in popular.values()) {
+            popScores.push(score);
+        }
+        const max = 5;
+        if (popScores.length < 5) {
+            max = popScores.length;
+        } 
+        for (var i = 0; i < max; i++) {
+            const highestScore = Math.max(popScores);
+            popScores.remove(highestScore);
+            const restaurantName = popular.getKey(highestScore);
+            popularList.appendChild(createPopularElement(restaurantName, highestScore));
+        }
+    });
+}
+
+// Creates a Popular List element
+function createPopularElement(restaurant, score) {
+  const popElement = document.createElement('a');
+  popElement.innerText = restaurant + "; Favorited by " + score + " users";
+  popElement.addEventListener("click", openModal(), false);
+  return popElement;
```

Comment:
sometimes you use 2 space while using tabs on others. can we be consistent?

Reply:
oh yeah my bad